### PR TITLE
Use an appropriate prompt based on the star rating

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-frontend/issues/1366

### CODE REVIEW

File: src/amo/components/AddonReview.js [hunk 1]
```diff
@@ -83,25 +83,38 @@ export class AddonReviewBase extends React.Component {
       );
     }
 
+    let placeholder;
+    let prompt;
+    if (review.rating && (review.rating > 3)) {
```
Comment (tofumatt) [inline src/amo/components/AddonReview.js]: The braces here make me expect something more and could be removed.

File: src/amo/components/AddonReview.js [hunk 2]
```diff
@@ -83,25 +83,38 @@ export class AddonReviewBase extends React.Component {
...
+      );
+      placeholder = i18n.gettext(
+        'Tell us what you love about this extension. Be specific and concise.'
+      );
+    } else {
+      prompt = i18n.gettext(
+        'Tell the world about this extension.'
+      );
+      placeholder = i18n.gettext(
+        'Tell us about your experience with this extension. ' +
```
Comment (tofumatt) [inline src/amo/components/AddonReview.js]: We should see about using the linter's `singleLineString` here (https://muffinresearch.co.uk/removing-leading-whitespace-in-es6-template-strings/)

File: src/amo/components/AddonReview.js (hunk 1 above)
Comment (kumar303) [inline src/amo/components/AddonReview.js]: what do you mean by braces? Did you mean parenthesis?

File: src/amo/components/AddonReview.js (hunk 2 above)
Comment (tofumatt) [inline src/amo/components/AddonReview.js]: It's a template string, so I think it does? We wrapped calls in the linter with a dummy gettext but never did extractions, so I'm unsure.

File: src/amo/components/AddonReview.js (hunk 1 above)
Comment (tofumatt) [inline src/amo/components/AddonReview.js]: Yeah, sorry. My brain got jumbled I meant:

```js
if (review.rating && review.rating > 3) { 
```
would work and I think is easier to read. I prefer the parens for more complicated statements inside `if`s.

Comment (kumar303) [inline src/amo/components/AddonReview.js]: Alright. I was nervous about the `&&` being applied first but `5 && 5 = 5` so I guess that's ok!

File: src/amo/components/AddonReview.js (hunk 2 above)
Comment (kumar303) [inline src/amo/components/AddonReview.js]: it looks like `dedent` will work: https://github.com/mozilla/addons-frontend/blob/master/src/disco/containers/DiscoPane.js#L99

Comment (tofumatt) [inline src/amo/components/AddonReview.js]: Aha, it seems they beat us to the "publish to npm bit", yeah that's totally what we were doing in the linter, we just rolled our own (oops).

👍

Comment (kumar303) [inline src/amo/components/AddonReview.js]: I don't think it's really possible here. Even with `dedent`, you can't create a space at the break which is necessary for the placeholder to read correctly in the textarea.

**Comment by tofumatt** [comment]
Oh, that’s true.


Okay, anything, was just a nitpick, it’s cool as is!


On 10 November 2016 at 22:55:30, Kumar McMillan (notifications@github.com)
wrote:

*@kumar303* commented on this pull request.
------------------------------

In src/amo/components/AddonReview.js
<https://github.com/mozilla/addons-frontend/pull/1368>:


> @@ -83,25 +83,38 @@ export class AddonReviewBase extends React.Component {
       );
     }

+    let placeholder;
+    let prompt;
+    if (review.rating && (review.rating > 3)) {
+      prompt = i18n.gettext(
+        'Tell the world why you think this extension is fantastic!'
+      );
+      placeholder = i18n.gettext(
+        'Tell us what you love about this extension. Be specific and concise.'
+      );
+    } else {
+      prompt = i18n.gettext(
+        'Tell the world about this extension.'
+      );
+      placeholder = i18n.gettext(
+        'Tell us about your experience with this extension. ' +

I don't think it's really possible here. Even with dedent, you can't create
a space at the break which is necessary for the placeholder to read
correctly in the textarea.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub
<https://github.com/mozilla/addons-frontend/pull/1368>, or mute the thread

**Comment by kumar303** [comment]
Thanks for the late night review 🍻

**Pull request merged**